roxie/roxiemem/roxiemem.cpp

```diff
@@ -2764,6 +2764,7 @@ void ChunkedHeaplet::verifySpaceList()
 
 char * ChunkedHeaplet::allocateChunk()
 {
+    //The spin lock for the heap this chunk belongs to must be held when this function is called
```

Comment:
Should we, in DEBUG or unittest mode only, add a test to see if the lock for this heap is already set ?